src/auro-accordion.js

```diff
@@ -81,79 +74,53 @@ class AuroAccordion extends LitElement {
   }
 
   /**
-   * @private Internal function to handle the click event to trigger the expansion of the accordion
-   * @param {object} event - Standard event parameter
+   * Toggles the visibility of the accordion content.
+   * @returns {void}
    */
-  handleClick(event) {
-    const nextState = !this.expanded;
-
-    this.expanded = nextState;
-
-    this.dispatchEvent(new CustomEvent('toggleExpanded', {
-      bubbles: true,
-      composed: true,
-      target: event.target
-    }));
+  toggle() {
+    this.expanded = !this.expanded;
   }
 
   /**
-   * @private Internal function to set a fixed height in slot content panel based on size of slot content. Necessary because CSS transitions won't work with height: auto.
+   * Used to generate inline style heights of content so it animates correctly.
+   * @private
+   * @returns {void}
    */
-  setHeight() {
-    const contentElemPanel = this.shadowRoot.querySelector(`#${this.id}Panel`);
-    const contentElem = contentElemPanel.querySelector(`.detailsSlot`);
-    const contentElemHeight = contentElem.offsetHeight;
+  handleContentSlotChanges() {
+    const content = this.shadowRoot.querySelector(".content");
+    const container = this.shadowRoot.querySelector(".contentWrapper");
+    const height = container.offsetHeight;
 
-    contentElemPanel.style.height = `${contentElemHeight}px`;
+    content.style.height = `${height}px`;
   }
 
   // function that renders the HTML and CSS into  the scope of the component
   render() {
-    const triggerStyles = {
-      'detailsTrigger': true,
-      'detailsTrigger--isExpanded': this.expanded
-    }
 
-    const detailStyles = {
-      'details': true,
-      'details--isExpanded': this.expanded,
-    }
+    const chevronHtml = this.chevron
+      ? html`
+          <div class="iconWrapper" part="chevron">
+            ${this.generateIconHtml(this.expanded ? chevronUp.svg : chevronDown.svg)}
+          </div>`
+      : html``
 
     return html`
-      <button
-        id="${this.id}Heading"
-        class="${classMap(triggerStyles)}"
-        aria-expanded="${this.expanded ? 'true' : 'false'}"
-        aria-controls="${this.id}Panel"
-        @click=${this.handleClick}>
-        <div class="triggerContentWrapper">
-          <div class="triggerContent">
-            <slot name="trigger" class="trigger"></slot>
-            <slot class="subTrigger" name="subTrigger"></slot>
+      <div class="componentWrapper">
```

Comment:
There is a csspart attribute on the chevron, but shouldn't other cssparts be added to the contents of the accordion as well?